presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveModule.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.airlift.configuration.AbstractConfigurationAwareModule;
+import com.facebook.presto.hive.HiveClientConfig;
+import com.facebook.presto.hive.MetastoreClientConfig;
+import com.facebook.presto.hive.PartitionMutator;
+import com.facebook.presto.hive.metastore.CachingHiveMetastore;
+import com.facebook.presto.hive.metastore.ExtendedHiveMetastore;
+import com.facebook.presto.hive.metastore.HiveMetastoreModule;
+import com.facebook.presto.hive.metastore.HivePartitionMutator;
+import com.facebook.presto.hive.metastore.MetastoreConfig;
+import com.google.inject.Binder;
+import com.google.inject.Scopes;
+
+import java.util.Optional;
+
+import static com.facebook.airlift.configuration.ConfigBinder.configBinder;
+
+public class IcebergHiveModule
+        extends AbstractConfigurationAwareModule
+{
+    private final String connectorId;
+    private final Optional<ExtendedHiveMetastore> metastore;
+
+    public IcebergHiveModule(String connectorId, Optional<ExtendedHiveMetastore> metastore)
+    {
+        this.connectorId = connectorId;
+        this.metastore = metastore;
+    }
+
+    @Override
+    public void setup(Binder binder)
+    {
+        install(new HiveMetastoreModule(this.connectorId, this.metastore));
+        install(new IcebergMetastoreModule());
```

Comment:
@ChunxuTang as I was going to separate some of these dependencies, I noticed this `IcebergMetastoreModule` class is never used after this binding. Pinging you to get your perspective on whether we should remove the class/file altogether or leave it as plumbing for a later feature 🙂